app/src/pages/playground/InvocationParameterJsonEditor.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useState } from "react";
+import { css } from "@emotion/react";
+
+import { Field } from "@arizeai/components";
+
+import { CodeWrap, JSONEditor } from "@phoenix/components/code";
+import { safelyParseJSON, safelyStringifyJSON } from "@phoenix/utils/jsonUtils";
+
+/**
+ * Editor for message tool calls
+ */
+export function InvocationParameterJsonEditor({
+  initialValue,
+  onChange: handleChange,
+  label,
+}: {
+  initialValue: unknown;
+  // jsonValue is typed as any on the gql side to allow any json value to be set
+  // we can mostly assume this any is safe because we are safely parsing and stringifying incoming and outgoing json
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onChange: (value: any) => void;
+  label: string;
+}) {
+  const [editorValue, setEditorValue] = useState(
+    initialValue == null ? "" : (safelyStringifyJSON(initialValue) ?? "")
+  );
+
+  const onChange = useCallback(
+    (value: string) => {
+      setEditorValue(value);
+      const { json: parsedValue } = safelyParseJSON(value);
+      if (parsedValue == null && value !== "") {
+        return;
+      }
+      handleChange(parsedValue);
+    },
+    [handleChange]
+  );
+
+  return (
+    <div
+      css={css`
+        & .ac-view {
+          width: 100%;
+          z-index: 1000;
+        }
```

Comment:
was the view rendering below the dialog or something? that feels strange to me we have plenty of forms in modals / slideovers i think

Reply:
Oops, meant to get rid of that, good catch. I was trying to fix clipped tooltips from codemirror when the editor is only 1 line tall.